test/utils.js

```diff
@@ -46,6 +46,23 @@ async function isUserInGroup(groupUser, group) {
   return userPresent;
 }
 
+async function waitTillUserInGroup(user, group, condition) {
+  let userInGroup = await isUserInGroup(user, group);
+  let timeOut = 0;
+  while (userInGroup !== condition) {
+    userInGroup = await isUserInGroup(user, group);
+    if (userInGroup === condition) {
+      return userInGroup;
+    }
+
+    await delay(1000);
+    timeOut++;
+    if (timeOut === 30) {
```

Comment:
Yes, discussed this with @bdemers as well. In the worst case, we wait for 30 seconds (which sucks) for the group rule to execute. I have never seen it take so long but if it does take so long, it might very well fail. It mostly takes 1~2 seconds to execute. 
Also, while using the mock server, the result is instantaneous and we don't have to wait.